### ISSUE TITLE
Error on joingenes step

### ISSUE THREAD

**Issue by etvedte** [Gaius-Augustus/BRAKER#170]
I'm running BRAKER in etp mode with evidence from RNA and protein. Here is the command (using generalized filenames):

/usr/local/packages/braker-2.1.2/braker.pl --species=DrosophilaAnanassae --softmasking --genome=fasta.masked --bam=sorted.bam --prot_seq=Dmel.uniprot.proteins.fasta --etpmode --prg=gth --gth2traingenes --cores=8 

The program produces some expected output from the documentation, i.e. augustus.hints.gtf, GeneMark-ETP/genemark.gtf, hintsfile.gff which have data in them, although no braker.gtf was produced.

The stderr of the job is below:

ERROR in file /usr/local/packages/braker-2.1.2/braker.pl at line 7986
Failed to execute: perl /usr/local/packages/braker-2.1.2/findGenesInIntrons.pl --in_gff=/autofs/projects-t3/RDBKO/dana.postassembly/braker/FREEZE/braker_out/augustus.Ppri5.gtf --jg_gff=/autofs/projects-t3/RDBKO/dana.postassembly/braker/FREEZE/braker_out/join.gtf --out_gff=/autofs/projects-t3/RDBKO/dana.postassembly/braker/FREEZE/braker_out/missed.genes.gtf 1> /dev/null 2> /autofs/projects-t3/RDBKO/dana.postassembly/braker/FREEZE/braker_out/errors/findGenesInIntrons.err

missed genes.gtf is populated, and there is nothing in findGenesInIntrons.err. 

I've attached the braker.log file. Do you have any idea of why these error messages are being generated? Although I do have output files, I want to be cautious about using them if they weren't properly generated...

Thanks in advance!
[braker.log](https://github.com/Gaius-Augustus/BRAKER/files/4236448/braker.log)

**Comment by KatharinaHoff** [comment]
Please update your BRAKER version.

Katharina

**Comment by KatharinaHoff** [comment]
The BRAKER call that you tried to use is actually not supposed to work. I have not implemented a check in braker.pl that will in the future abort runs with --etpmode and GenomeThreader.

https://github.com/Gaius-Augustus/BRAKER/commit/7adbae1f32c8cec90e8a5fccaaab4772a078aff4

**Comment by etvedte** [comment]
Thank you for the prompt response. I will test and check back.

**Comment by etvedte** [comment]
Katharina,

I've updated the software and performed a successful run of BRAKER with --prg=gth --gth2traingenes. I have not yet tested with --etpmode.

I am a little confused by the tool documentation, which states gene prediction/training with genomethreader via --prg=gth + --gth2traingenes is a deprecated pipeline, while GeneMark-ETP training is "only available as a premature version." What is the recommended practice for training/predicting when both RNAseq (from the same species) and protein hints (from a closely related species) are available? 

Thanks for your feedback.

**Comment by KatharinaHoff** [comment]
I will remove the "premature version" ;-) Sorry for lagging behind.
We definitely recommend using ProtHint/GeneMark-EP, now.

Katharina

**Comment by etvedte** [comment]
OK, thanks again for your prompt responses. 

Cheers,

Eric

**Comment by KatharinaHoff** [comment]
Sorry, that was a misunderstanding on my side. There are several things:

BRAKER --epmode: This is what we recommend for training AUGUSTUS within BRAKER using proteins of any evolutionary distance.

BRAKER --etpmode: This is a premature pipeline. It can be used, but on the poster from PAG 2020, you will find that it does not always perform better than using protein or RNA-Seq alone. However, if you want to mix RNA-Seq and proteins, anyway, use this. The pipeline is subject to constant changes, and I expect largest improvements in BRAKER in --etpmode within the next year.

BRAKER with GenomeThreader: we recommended this while ProtHint wasn't available, yet. I currently do not recommend it and will not further maintain it.

BRAKER with GenomeThreader and RNA-Seq for training: was always unstable in accuracy across different species. I won't work on improving it.

Long story short: we currently cannot clearly recommend to mix RNA-Seq and protein data without manual checking whether that leads to an improvement or makes predictions worse.

**Comment by etvedte** [comment]
Katharina,

Per your recommendation, I tried out a new pipeline with --etpmode, having both RNAseq and protein data available. The pipeline runs for a while, then fails.

The error log for the BRAKER run reads:
```
Training gene file in genbank format /autofs/projects-t3/LGT/Dananassae_2020/dana.postassembly/braker/FREEZE/braker_out_etp/train.f.gb does not contain any training genes. Possible known cause: no training genes with sufficient extrinsic evidence support or of sufficient length were produced by GeneMark-ES/ET.
```
The pipeline seems to have issues at this step:
```
/usr/local/packages/augustus-3.3.3/scripts/filterGenesIn_mRNAname.pl
```
 Both good_genes.lst and train.gb have data and have matching names in "transcript_id," but for some reason these matches aren't populating the train.f.gb file.

This seems similar to previous issues.  

https://github.com/Gaius-Augustus/BRAKER/issues/150
https://github.com/Gaius-Augustus/BRAKER/issues/162

Per the documentation's recommendation, we installed AUGUSTUS 3.3.3 via git clone. The pipeline also failed at this step with only RNAseq data and without using --etpmode

**Comment by Chenglin20170390** [comment]
HI etvedte , I figured out this problem by changing Augustus-3.3.3/scripts/filterGenesIn_mRNAname.pl 
reference [https://github.com/Gaius-Augustus/BRAKER/issues/126](url)
`# instead of
f ( $_ =~ m/transcript_id "(.*)";/ ) {

#simply adding a questionmark (?)
f ( $_ =~ m/transcript_id "(.*?)";/ ) {`

**Comment by KatharinaHoff** [comment]
filterGenesIn_mRNAname.pl has changed (to a different regular expression) and should not produce this problem, anymore.

https://github.com/Gaius-Augustus/Augustus/commit/c21845d95a9928afa6c6ed5e7a41a4a3ada9eac2
